### PR TITLE
chore(tests) use 3.x release images

### PR DESCRIPTION
**Pull request by rainest**
**What this PR does / why we need it**:

Updates the 3.x compatibility test to use release images. Awaiting https://github.com/Kong/kubernetes-ingress-controller/actions/runs/3039067881 as a test run against this branch.

Changes the the various `/bin/sh -c` invocations in container commands to `/bin/bash -c` as all images provide a bash and the default's `/bin/sh` is not compatible with our scripts.

Fixes the error type check for Knative's IngressClass scan; I forgot it earlier when I fixed the not-Knative version.

**Which issue this PR fixes**:
The Enterprise image pull credentials were not properly configured; this fixes those by virtue of no longer requiring them. We want to start using the release images anyway since they're now the newest versions.

However, it breaks the OSS runs because that image isn't actually available. I'm okay with this; they were passing before, and they will need to pass with that image eventually. This version turns the existing compatibility test into a trial run for the release E2E tests, which will require the released image.

### CODE REVIEW

**Comment by rainest** [comment]
Didn't work. Docker official library image isn't out yet. Not sure what broke the Enterprise one, lacking diagnostics on anything that failed.

**Comment by rainest** [comment]
Postgres tests for Enterprise are failing with https://gist.github.com/rainest/e5aaf08e1e4da19b1fd3dc26520f7170 where the initial migrations job will log
```
openid-connect migrated up to: 000_base_openid_connect (executed)
openid-connect migrated up to: 001_14_to_15 (executed)
openid-connect migrated up to: 002_200_to_210 (executed)
openid-connect migrated up to: 003_280_to_300 (executed)
```
and then the attempt to kong start after will fail with
```
Pending migrations:
openid-connect: 002_200_to_210
New migrations available:
                          openid-connect: 003_280_to_300
...
```
so apparently one of those is lying. Kong never comes up and the tests never really start, but not through obvious fault of the Deployment.

Some other failures stem from the removal of `-enterprise-edition` from the version string, since E2E had its own version sniffer.

**Comment by rainest** [comment]
Migrations errors appear to have been a bit misleading and are complaining before migrations finish (this is normal) but not succeeding after (this is not). This is caused by Debian's `/bin/sh` (now the default image) not accepting the syntax we use, so it can never check the return value:

```
$ docker run --rm kong/kong-gateway:3.0 /bin/sh -c '[[ 1 ]] && echo "it works"'
/bin/sh: 1: [[: not found
```

bash does accept it and is available on all our platforms (though sometimes it's BusyBox IIRC):

```
17:05:43-0700 esenin $ docker run --rm kong/kong-gateway:3.0.0.0-alpine /bin/bash -c "echo aaa"                                                                                       
aaa
17:09:48-0700 esenin $ docker run --rm kong/kong-gateway:3.0.0.0-debian /bin/bash -c "echo aaa"  
aaa
17:09:55-0700 esenin $ docker run --rm kong/kong-gateway:3.0.0.0-ubuntu /bin/bash -c "echo aaa"  
aaa
17:09:58-0700 esenin $ docker run --rm kong/kong-gateway:3.0.0.0-rhel /bin/bash -c "echo aaa"  
aaa
```

https://github.com/Kong/kubernetes-ingress-controller/actions/runs/3041434734 is currently running with a fixed version tester and a change to bash. May yet uncover new problems. OSS will still fail since that image is missing, but I'm less concerned with it since it was passing during earlier scheduled runs.

**Comment by rainest** [comment]
Failed because these manifests were still using busybox (the image, not the binary), which does not include bash. Now trying with the Kong image instead: https://github.com/Kong/kubernetes-ingress-controller/actions/runs/3041700933

We already replaced busybox with the Kong image long ago when we made a similar change to bash from sh in the chart; probably should have changed this then too.

File: deploy/single/all-in-one-postgres.yaml [hunk 1]
```diff
@@ -1805,7 +1805,7 @@ spec:
           value: postgres
         - name: KONG_PG_PORT
           value: "5432"
-        image: busybox
+        image: kong:2.8
```
Comment (randmonkey) [inline deploy/single/all-in-one-postgres.yaml:1808]: `kong/kong-gateway:3.0` does not contain `nc` command, so we need another command to test connectivity of DB.

**Review by randmonkey — CHANGES_REQUESTED** [review]

**Review by randmonkey — APPROVED** [review]

**Pull request merged**